packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/log_stream_provider.dart

```diff
@@ -36,15 +40,22 @@ class DefaultCloudWatchLogStreamProvider
   final String _logGroupName;
   final CloudWatchLogsClient _client;
   static final DateFormat _dateFormat = DateFormat('yyyy-MM-dd');
+  final _cachedLogStreams = <String>{};
```

Comment:
Maybe `_createdLogStreams` is a better name since your keeping track of what's been created.